src/main/scala/de/m7w3/signal/messages/MessageReceiver.scala

```diff
@@ -51,17 +50,16 @@ case class MessageReceiver(cipher: SignalServiceCipher,
     try {
       val envelope = pipe.read(timeoutMillis, TimeUnit.MILLISECONDS)
 
-      if (envelope.isPreKeySignalMessage) {
-        logger.debug(s"got prekeys from ${envelope.getSourceAddress} ${envelope.getSourceDevice}")
-        // TODO: handle
-      } else if (envelope.isReceipt) {
-        logger.debug(s"got receipt from ${envelope.getSourceAddress} ${envelope.getSourceDevice}")
+      if (envelope.isReceipt) {
+        logger.debug(s"received receipt from ${envelope.getSource}")
+        // not handled yet, just an example
+        eventPublisher.publishEvent(ReceiptEvent(envelope.getSourceAddress, envelope.getSourceDevice, envelope.getTimestamp))
```

Comment:
tip if I may: You can create an alternative constructor for `ReceiptEvent` which takes an `Envelope` and does the fields extractions for you. So that when we need to create `ReceiptEvent` several times, we don't have to get the fields each time